tinygrad/tensor.py

```diff
@@ -927,11 +927,11 @@ def layernorm(self, axis=-1, eps:float=1e-5) -> Tensor:
     y = (self - self.mean(axis, keepdim=True))
     return y.mul((y*y).mean(axis, keepdim=True).add(eps).rsqrt())
 
-  def batchnorm(self, weight:Optional[Tensor], bias:Optional[Tensor], mean:Tensor, invstd:Tensor) -> Tensor:
-    shape = (1, -1) + (1,) * (self.ndim-2)
+  def batchnorm(self, weight:Optional[Tensor], bias:Optional[Tensor], mean:Tensor, invstd:Tensor, axis:Union[int,Tuple[int,...]]=1) -> Tensor:
+    shape = tuple(s if ax in argfix(axis) else 1 for ax, s in enumerate(self.shape))
```

Comment:
Do `argfix(axis)` first so it's not being called multiple time. Also can you add a direct unit test for case when axis is a tuple in test_ops?